src/pilot_simulation.jl

```diff
@@ -123,23 +130,214 @@ forcings = ModelForcing(u=Fu′, v=Fv′, w=Fw′, T=Fθ′, S=Fs′)
+    "κS" => Dict("longname" => "Eddy diffusivity of absolute salinity", "units" => "m²/s"),
+    "uu" => Dict("longname" => "Velocity covariance between u and u", "units" => "m²/s²"),
+    "vv" => Dict("longname" => "Velocity covariance between v and v", "units" => "m²/s²"),
+    "ww" => Dict("longname" => "Velocity covariance between w and w", "units" => "m²/s²"),
+    "uv" => Dict("longname" => "Velocity covariance between u and v", "units" => "m²/s²"),
+    "uw" => Dict("longname" => "Velocity covariance between u and w", "units" => "m²/s²"),
+    "vw" => Dict("longname" => "Velocity covariance between v and w", "units" => "m²/s²"),
+    "wT" => Dict("longname" => "Vertical turbulent heat flux", "units" => "K*m/s"),
+    "wS" => Dict("longname" => "Vertical turbulent salinity flux", "units" => "g/kg*m/s")
+)
+
+#####
+##### Fields output writer
+#####
+
+fields = Dict(
+    "u"  => model.velocities.u,
+    "v"  => model.velocities.v,
+    "w"  => model.velocities.w,
+    "T"  => model.tracers.T,
+    "S"  => model.tracers.S,
+    "ν"  => model.diffusivities.νₑ,
+    "κT" => model.diffusivities.κₑ.T,
+    "κS" => model.diffusivities.κₑ.S
+)
+
+field_output_writer =
+    NetCDFOutputWriter(model, fields, filename=filename_prefix * "_fields.nc", interval=6hour,
+                      global_attributes=global_attributes, output_attributes=output_attributes)
+
+#####
+##### Horizontal averages output writer
+#####
+
+profiles = Dict(
+    "u"  => model ->  Up(model)[1+model.grid.Hz:end-model.grid.Hz],
+    "v"  => model ->  Vp(model)[1+model.grid.Hz:end-model.grid.Hz],
+    "w"  => model ->  Wp(model)[1+model.grid.Hz:end-model.grid.Hz],
+    "T"  => model ->  Tp(model)[1+model.grid.Hz:end-model.grid.Hz],
+    "S"  => model ->  Sp(model)[1+model.grid.Hz:end-model.grid.Hz],
+    "ν"  => model ->  νp(model)[1+model.grid.Hz:end-model.grid.Hz],
+    "κT" => model -> κTp(model)[1+model.grid.Hz:end-model.grid.Hz],
+    "κS" => model -> κSp(model)[1+model.grid.Hz:end-model.grid.Hz],
+    "uu" => model ->  uu(model)[1+model.grid.Hz:end-model.grid.Hz],
+    "vv" => model ->  vv(model)[1+model.grid.Hz:end-model.grid.Hz],
+    "ww" => model ->  ww(model)[1+model.grid.Hz:end-model.grid.Hz],
+    "uv" => model ->  uv(model)[1+model.grid.Hz:end-model.grid.Hz],
+    "uw" => model ->  uw(model)[1+model.grid.Hz:end-model.grid.Hz],
+    "vw" => model ->  vw(model)[1+model.grid.Hz:end-model.grid.Hz],
+    "wT" => model ->  wT(model)[1+model.grid.Hz:end-model.grid.Hz],
+    "wS" => model ->  wS(model)[1+model.grid.Hz:end-model.grid.Hz]
+)
+
+profile_dims = Dict(k => ("zC",) for k in keys(profiles))
+profile_dims["w"] = ("zF",)
+
+profile_output_writer =
+    NetCDFOutputWriter(model, profiles, filename=filename_prefix * "_profiles.nc", interval=10minute,
+                      global_attributes=global_attributes, output_attributes=output_attributes,
+                      dimensions=profile_dims)
+
+#####
+##### Large scale solution output writer
```

Comment:
Large scale = 3D fields?

Reply:
Ah sorry it's the base state so really just vertical profiles as a function of time.